src/main/java/de/uos/informatik/ko/rcp/geneticAlgo/GeneratePop.java

```diff
@@ -0,0 +1,101 @@
+        // all activities without predecessor are possible candidates
+        HashSet<ArrayList<Integer>> results = new HashSet<ArrayList<Integer>>();
+        HashMap<Integer,HashSet<Integer>> mypredecessors;
+        HashSet<Integer> candidates = new HashSet<Integer>();
+        //scheduled activities
+        ArrayList<Integer> scheduled = new ArrayList<Integer>();
+        for (int i=0; i< pop_size*5; i++) {
+            mypredecessors = Utils.buildPredecessorMap(instance);
+            for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                if (entry.getValue().isEmpty() && (!(scheduled.contains(entry.getKey())))) {
+                    candidates.add(entry.getKey());
+                }
+            }
+            Integer mycandidate = chooseone(candidates, myGenerator);
+            scheduled.add(mycandidate);
+            candidates.remove(mycandidate);
+            //delete scheduled activity from predecessor list
+            for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                entry.getValue().remove(mycandidate);
+            }
+            while (scheduled.size() < instance.n()) {
+                for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                    if (entry.getValue().isEmpty()&& (!(scheduled.contains(entry.getKey())))) {
+                        candidates.add(entry.getKey());
+                    }
+                }
+                Integer actcandidate = chooseone(candidates, myGenerator);
+                scheduled.add(actcandidate);
+                candidates.remove(actcandidate);
+                // delete scheduled activity from predecessor list
+                for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                    entry.getValue().remove(actcandidate);
+                }
+            }
+            ArrayList<Integer> resultsList = new ArrayList<Integer>();
+            for (Integer elem: scheduled){
+                resultsList.add(elem);
+            }
+            results.add(resultsList);
+            scheduled.clear();
+            candidates.clear();
+            if (results.size() >= pop_size){
+                break;
+            }
+        }
+        return results;
+    }
+    // chooses randomly one Integer out of the Integer in the given Set, the Random Generator is given
+    public static Integer chooseone( HashSet<Integer> candidates, Random myGenerator){
+        Integer returncanidate;
+        int mycandidate=0;
+        int item = myGenerator.nextInt(candidates.size());
+        int i = 0;
+        for(Integer elem: candidates) {
+            if (i == item) {
+                mycandidate = elem;
+                break;
+            }
+            i++;
+        }
+        returncanidate = mycandidate;
+        return returncanidate;
```

Comment:
I like the above mentioned version from stackoverflow.

Concerning your second comment, i think we are loosing the possibility to use the seed, since it isn't used if we just use the iterator right? 
I think it could be helpful if we try to test or debug our code.
And we have, as you mentioned, to test how "deterministic" the iterator is.

Reply:
Good point about using a seed, didn't consider that :+1:  We should stick to the SO version then IMO